lib/src/exceptions.dart

```diff
@@ -6,29 +6,43 @@
 /// Thrown when there is an error during the authentication flow.
 class DRAWAuthenticationError implements Exception {
   DRAWAuthenticationError(this.message);
-  String message;
+  final String message;
   String toString() => 'DRAWAuthenticationError: $message';
 }
 
-/// Thrown by unfinished code that hasn't yet implemented all the features it
-/// needs.
-class DRAWUnimplementedError extends UnimplementedError {
-  DRAWUnimplementedError([String message]) : super(message);
+/// Thrown due to invalid arguments being provided to a DRAW method.
+class DRAWArgumentError implements Exception {
+  DRAWArgumentError(this.message);
+  final String message;
+  String toString() => 'DRAWArgumentError: $message';
 }
 
 /// Thrown due to a fatal error encountered inside DRAW. If you're not adding
 /// functionality to DRAW you should never see this. Otherwise, please file a
 /// bug at github.com/draw-dev/DRAW/issues.
 class DRAWInternalError implements Exception {
   DRAWInternalError(this.message);
-  String message;
+  final String message;
   String toString() => 'DRAWInternalError: $message';
 }
 
 /// Thrown due to a error on the side of the client due to incorrect integration of DRAW.
 class DRAWClientError implements Exception {
   DRAWClientError(this.message);
-  String message;
+  final String message;
   String toString() =>
       'DRAWClientError: $message This is likely due to issues with your ini file.';
 }
+
+class DRAWRedirectResponse implements Exception {
```

Comment:
Should we call this Exception type: DRAWRedirectResponseError instead?

Reply:
This isn't an error, but is still an exception, so we should leave it as is.